lib/galaxy/jobs/handler.py

```diff
@@ -965,8 +965,12 @@ def recover(self, job, job_wrapper):
             job_wrapper.fail(DEFAULT_JOB_PUT_FAILURE_MESSAGE)
 
     def shutdown(self):
-        for runner in self.job_runners.values():
+        for name, runner in self.job_runners.items():
+            failures = []
```

Comment:
Was the idea to collect all names fo all job_runners before raising the Exception? If so should this not be outside the for loop?

Reply:
Indeed, thanks!